File: Movil/mobile/src/main/java/com/example/movil/models/Sesion.kt
package com.example.movil.models

class Sesion(
    val sesionId: Int,
    val token: String?,
    val inicia: String,
    val finaliza: String?,
    val usuarioId: Int,
) {
    constructor(data: Any?) : this(
        sesionId = ((data as? Map<*, *>)?.get("sesionId") as? Double)?.toInt() ?: 0,
        token = (data as? Map<*, *>)?.get("token") as? String,
        inicia = (data as? Map<*, *>)?.get("inicia") as? String ?: "",
        finaliza = (data as? Map<*, *>)?.get("finaliza") as? String,
        usuarioId = ((data as? Map<*, *>)?.get("usuarioId") as? Double)?.toInt() ?: 0
    )

}

File: Servicios/Models/Tables/Sesion.cs
using System.ComponentModel.DataAnnotations;

namespace Servicios.Models.Tables
{
    public class Sesion
    {
        [Key]
        public int SesionId { get; set; }
        public string? Token { get; set; }
        public DateTime Inicia { get; set; }
        public DateTime? Finaliza { get; set; }
        public int UsuarioId { get; set; }
        public virtual Usuario Usuario { get; set; }
    }
}
